obs2ioda-v2/src/ncio_mod.f90

```diff
@@ -176,90 +207,115 @@ subroutine write_obs (filedate, write_opt, outdir, itim)
          end if
          if ( iflag /= itrue ) cycle var_info_def_loop
          ncname = trim(name_var_info(i))
-         igrp = ufo_vars_getindex(name_ncgrp, 'MetaData')
-         idim = ufo_vars_getindex(name_ncdim, dim_var_info(1,i))
-         dim1 = ncid_ncdim(idim)
          if ( ufo_vars_getindex(name_ncdim, dim_var_info(2,i)) > 0 ) then
+            idim = ufo_vars_getindex(name_ncdim, dim_var_info(1,i))
+            dim1 = ncid_ncdim(idim)
+            dim1_name = get_dim_name(ncid_ncdim(dim1), nchans_nvars_flag)
```

Comment:
Can these three lines be factored out of the if-else blocks here and moved above the if-test at line 210? If so, that would more closely match the logic in the original code.